jwplatform/upload.py

```diff
@@ -0,0 +1,132 @@
+import os
+from enum import Enum
+from hashlib import md5
+import requests
+from requests import HTTPError
+
+from jwplatform import constants
+
+
+class UploadType(Enum):
+    direct = "direct"
+    multipart = "multipart"
+
+
+def determine_upload_method(file, target_part_size) -> str:
+    filename = file.name
+    file_size = os.stat(filename).st_size
```

Comment:
It might be worth doing something like this: https://stackoverflow.com/a/19079887, mostly so that users could pass "file-like" objects such as streams